packages/@ama-sdk/core/src/plugins/session-id/session-id.request.ts

```diff
@@ -32,29 +33,40 @@ export class SessionIdRequest implements RequestPlugin {
   /**
    * The session ID used to track API calls
    */
-  public sessionId: string;
+  public sessionId!: string;
 
   /**
    * Constructor.
    * @param sessionIdHeader The request header in which the ID will be added.
    * @param activateRequestId Indicates if the request ID part should be added to the ID.
    */
   constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true) {
+    // Declaration done first since generateSessionId uses the logger
     this.sessionIdHeader = sessionIdHeader;
     this.requestIdActivated = activateRequestId;
-    this.sessionId = this.generateSessionId();
   }
 
-  private logSessionId(sessionId: string, date: string) {
-    // eslint-disable-next-line no-console, no-restricted-syntax
-    console.info(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
-    // eslint-disable-next-line no-console, no-restricted-syntax
-    console.info(`Generated at: ${date}`);
+  private logSessionId(sessionId: string, date: string, logger?: Logger) {
+    if (logger?.info) {
+      logger.info(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
+      logger.info(`Generated at: ${date}`);
+    } else if (logger?.log) {
+      logger.log(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
+      logger.log(`Generated at: ${date}`);
+    } else {
+      console.info(`Your debug ID associated to the header "${this.sessionIdHeader}" is: ${sessionId}.`);
+      console.info(`Generated at: ${date}`);
+    }
   }
 
-  public load(): PluginRunner<RequestOptions, RequestOptions> {
+  /**
+   * Load the plugin with the context
+   * @param context Context of request plugin
+   */
+  public load(context?: RequestPluginContext): PluginRunner<RequestOptions, RequestOptions> {
     return {
       transform: (data: RequestOptions) => {
+        this.sessionId = this.generateSessionId(context?.logger);
```

Comment:
why it is not in `load` instead of `transform` that would reduce the number of call?
Also you may want to use `||=` instead of `=` for the same purpose